sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java

```diff
@@ -248,16 +250,24 @@ private boolean isSideInputLookupJoin() {
       PCollection<Row> leftRows = pinput.get(0);
       PCollection<Row> rightRows = pinput.get(1);
 
+      int leftRowColumnCount = leftRelNode.getRowType().getFieldCount();
+
       // extract the join fields
-      List<Pair<Integer, Integer>> pairs =
-          extractJoinColumns(leftRelNode.getRowType().getFieldCount());
+      List<Pair<RexNode, RexNode>> pairs = extractJoinRexNodes();
 
       // build the extract key type
       // the name of the join field is not important
       Schema extractKeySchemaLeft =
-          pairs.stream().map(pair -> leftSchema.getField(pair.getKey())).collect(toSchema());
+          pairs.stream()
+              .map(pair -> BeamJoinRel.getFieldBasedOnRexNode(leftSchema, pair.getKey(), 0))
```

Comment:
you don't need the `BeamJoinRel` to reference the static method in this class

Reply:
Done